### PR TITLE
CSTA Form - Teacher Role and Grade Band - With DCDO

### PR DESCRIPTION
**Pull request by epeach**
A version of[ this PR](https://github.com/code-dot-org/code-dot-org/pull/49278), but with a DCDO flag, in case we decide that's safer.

Getting this prepped in parallel due to HoC code freeze time crunch.

### CODE REVIEW

File: apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx [hunk 1]
```diff
@@ -70,6 +87,16 @@ export default class AmazonFutureEngineerEligibilityForm extends React.Component
     this.setState(change);
   };
 
+  handleRoleChange = change => {
+    this.setState({professionalRole: change.target.value});
+  };
+
+  handleMultiSelectGradeBands = index => {
+    let gradeBands = this.state.gradeBands;
+    gradeBands[index] = !gradeBands[index];
```
Comment (bethanyaconnor) [inline apps/src/templates/amazonFutureEngineerEligibility/amazonFutureEngineerEligibilityForm.jsx:96]: Sorry, I should've caught this in the previous PR but updating arrays in state require a copy (more info in [React docs](https://beta.reactjs.org/learn/updating-arrays-in-state#updating-arrays-without-mutation)). I think you could do something like:
```
const gradeBands = {...this.state.gradeBands};
gradeBands[index] = !gradeBands[index];
this.setState(gradeBands);
```

File: dashboard/lib/services/csta_enrollment.rb [hunk 2]
```diff
@@ -51,7 +53,9 @@ def self.submit(first_name:, last_name:, email:, school_district_name:,
         "submission[17_city]" => city.titleize,
         "submission[17_state]" => get_us_state_abbr(state, true),
         "submission[17_zip]" => zip,
-        "submission[19]" => "Yes, I provide my consent."
+        "submission[19]" => "Yes, I provide my consent.",
+        "submission[20]" => professional_role,
```
Comment (bethanyaconnor) [inline dashboard/lib/services/csta_enrollment.rb:57]: Just confirming: if the dcdo flag is off, these will be `nil` and the submission will be like we didn't even try to set these key-values right?

Comment (epeach) [reply dashboard/lib/services/csta_enrollment.rb:57]: That's the idea! If my ruby is correct, this should be handled in `amazon_future_engineer_controller`, we call to this function with ```professional_role: afe_params['primaryProfessionalRole'] || '',
        grades_teaching: afe_params['gradesTeaching'] || '',```.
        
So if there is no input there, we'll just pass along "".

**Review by bethanyaconnor — APPROVED** [review]

File: pegasus/sites.v3/code.org/views/amazon_future_engineer_eligibility.haml [hunk 3]
```diff
@@ -4,6 +4,8 @@
 
   window.dashboard = window.dashboard || {};
   window.dashboard.CODE_ORG_URL = "#{CDO.code_org_url}";
+-# data needed for dcdo.js
```
Comment (breville) [inline pegasus/sites.v3/code.org/views/amazon_future_engineer_eligibility.haml:8]: suggestion: a blank line before this.

**Pull request merged**